Tests/XRP/Fakes/FakeXRPClient.swift

```diff
@@ -71,6 +75,15 @@ extension FakeXRPClient: XRPClientDecorator {
     return try returnOrThrow(result: accountExistsValue)
   }
 
+  public func getPayment(for transactionHash: TransactionHash) throws -> XRPTransaction? {
+    switch getPaymentValue {
```

Comment:
You can just use the `returnOrThrow` helper method which does the same thing as your logic.

Reply:
so the reason I had to roll my own here is that `returnOrThrow` requires the type `Result<T, XRPLedgerError>`, but I needed to switch on type `Result<XRPTransaction?, RPCError>`